pkg/openthread/contrib/sock/udp/openthread_sock_udp.c

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <errno.h>
+
+#include "net/ipv6/addr.h"
+#include "net/sock/udp.h"
+#include "net/udp.h"
+#include "net/sock.h"
+#include "timex.h"
+#include "net/af.h"
+#include "ot.h"
+#include "openthread/thread.h"
+#include "openthread/udp.h"
+#include "byteorder.h"
+#include "net/protnum.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+static void _handle_receive(void *aContext, otMessage *aMessage, const otMessageInfo *aMessageInfo)
+{
+    size_t payload_len = otMessageGetLength(aMessage)-otMessageGetOffset(aMessage);
+    sock_udp_t *sock = aContext;
+	if(sock->data == NULL || payload_len > sock->max_len)
+	{
+		return;
+	}
+	
+    otMessageRead(aMessage, otMessageGetOffset(aMessage), sock->data, payload_len);
+	sock->data = NULL;
```

Comment:
there are some unfinished parts in this snippet. I will remove those when they are OK